### PR TITLE
Update node agent key/cert file to be the same as workload key/cert, …

### PR DESCRIPTION
**Pull request by wattli**
…extract requested identity from cert

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by wattli** [comment]
I am adding unti test for GetServiceIdentity now, but the logic is ready for review now.

**Comment by wattli** [comment]
I added a do-not-merge label and will not remove it until the unit test is ready

File: cmd/node_agent/na/gcp.go [hunk 1]
```diff
@@ -53,3 +53,7 @@ func (na *gcpPlatformImpl) GetDialOptions(cfg *Config) ([]grpc.DialOption, error
 	options := []grpc.DialOption{grpc.WithPerRPCCredentials(&jwtAccess{jwtKey})}
 	return options, nil
 }
+
+func (na *gcpPlatformImpl) GetServiceIdentity(file string) (string, error) {
+	return "", nil
```
Comment (lookuptable) [inline cmd/node_agent/na/gcp.go]: How do we decide the service identity in GCE?

File: cmd/node_agent/na/onprem_test.go~ [hunk 2]
```diff
@@ -0,0 +1,50 @@
+// Copyright 2017 Istio Authors
```
Comment (lookuptable) [inline cmd/node_agent/na/onprem_test.go~]: Please fix the file name here

File: cmd/node_agent/na/nodeagent.go [hunk 3]
```diff
@@ -167,8 +162,12 @@ func (na *nodeAgentInternal) Start() error {
 }
 
 func (na *nodeAgentInternal) createRequest() ([]byte, *pb.Request, error) {
+	identity, err := na.pr.GetServiceIdentity(na.config.CertChainFile)
```
Comment (lookuptable) [inline cmd/node_agent/na/nodeagent.go]: Is this something we can read once and set it into a field of nodeAgentInternal?

File: cmd/node_agent/na/onprem.go [hunk 4]
```diff
@@ -39,6 +40,25 @@ func (na *onPremPlatformImpl) IsProperPlatform() bool {
...
+		return "", err
+	}
+	cert, err := pki.ParsePemEncodedCertificate(certBytes)
+	if err != nil {
+		glog.Errorf("Failed to parse cert chain bytes: %s", err)
+		return "", err
+	}
+	serviceIDs := pki.ExtractIDs(cert.Extensions)
+	if len(serviceIDs) != 1 {
+		glog.Errorf("Cert have %v SAN fields, should be 1", len(serviceIDs))
```
Comment (lookuptable) [inline cmd/node_agent/na/onprem.go]: We should avoid both logging an error and returning en error. Doing this is likely to cause an error being logged more than once.

File: cmd/node_agent/na/gcp.go (hunk 1 above)
Comment (wattli) [reply cmd/node_agent/na/gcp.go]: It's undecided, in the jwt ideally.

File: cmd/node_agent/na/onprem_test.go~ (hunk 2 above)
Comment (wattli) [reply cmd/node_agent/na/onprem_test.go~]: Done.

File: cmd/node_agent/na/gcp.go (hunk 1 above)
Comment (wattli) [reply cmd/node_agent/na/gcp.go]: Leave a todo

File: cmd/node_agent/na/onprem.go (hunk 4 above)
Comment (wattli) [reply cmd/node_agent/na/onprem.go]: Done

File: cmd/node_agent/na/gcp.go (hunk 1 above)
Comment (lookuptable) [reply cmd/node_agent/na/gcp.go]: Then this shouldn't be a common method. In GCP we'll never find the identity from a file

File: cmd/node_agent/na/nodeagent.go (hunk 3 above)
Comment (wattli) [reply cmd/node_agent/na/nodeagent.go]: good point, done.

File: cmd/node_agent/na/gcp.go (hunk 1 above)
Comment (wattli) [reply cmd/node_agent/na/gcp.go]: Chat offline, done.

**Comment by wattli** [comment]
@lookuptable , please take a look

File: cmd/node_agent/na/testdata/cert-chain-bad1.pem [hunk 5]
```diff
@@ -0,0 +1,26 @@
+-----BEGIN CERTIFICATE-----
```
Comment (lookuptable) [inline cmd/node_agent/na/testdata/cert-chain-bad1.pem]: This file is not used anywhere, no?

Comment (wattli) [reply cmd/node_agent/na/testdata/cert-chain-bad1.pem]: Good point, done.

**Pull request merged**